nautobot/extras/models/groups.py

```diff
@@ -165,8 +165,15 @@ def _map_filter_fields(self):
         # Get dynamic group filter field mappings (if any)
         dynamic_group_filter_fields = getattr(self.model, "dynamic_group_filter_fields", {})
 
-        # Model form fields that aren't on the filter form
-        missing_fields = set(modelform_fields).difference(filterform_fields)
+        # Whether or not to add missing form fields that aren't on the filter form.
+        skip_missing_fields = getattr(self.model, "dynamic_group_skip_missing_fields", False)
+
+        # Model form fields that aren't on the filter form.
+        if not skip_missing_fields:
+            missing_fields = set(modelform_fields).difference(filterform_fields)
+        else:
+            logger.debug("Will not add missing form fields due to model meta option.")
+            missing_fields = set()
 
         # Try a few ways to see if a missing field can be added to the filter fields.
         for missing_field in missing_fields:
```

Comment:
Given `missing_fields` is a `set`, I wonder if we should be sorting it in some way before iterating over it, so as to ensure consistent ordering in the generated form?

Reply:
Yes. There was a lingering todo item to also augment/extend the form's existing field ordering and re-order the fields using [`Form.order_fields(field_order)`](https://docs.djangoproject.com/en/4.1/ref/forms/api/#django.forms.Form.order_fields)